src/ipc/handler.c

```diff
@@ -674,7 +674,8 @@ static int ipc_pm_gate(uint32_t header)
 
 	if (pm_gate.flags & SOF_PM_PPG)
 		pm_runtime_disable(PM_RUNTIME_DSP, PLATFORM_MASTER_CORE_ID);
-	else
+	else if (platform_pm_runtime_is_active(PM_RUNTIME_DSP,
```

Comment:
A similar issue is being solved by #3113 wrt HPRO requests in order to deal with unbalanced requests. There is a bitmap of clients requesting override. Similar thing would help here (dedicated bit for host_pg_override and another one for fw_pg_override). Then multiple unbalanced req/de-req does not cause issues.